contributors.d

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env rdmd
+
+/**
+Query contributors between two D releases.
+
+Copyright: D Language Foundation 2017.
+
+License:   $(WEB boost.org/LICENSE_1_0.txt, Boost License 1.0).
+
+Example usage:
+
+---
+./contributors.d "v2.074.0..v2.075.0"
+---
+
+Author: Sebastian Wilzbach
+*/
+
+import std.array;
+import std.algorithm;
+import std.conv;
+import std.exception;
+import std.file;
+import std.format;
+import std.process;
+import std.path;
+import std.range;
+import std.stdio;
+import std.string;
+import std.typecons;
+
+/// Name <my@email.com>
+struct GitAuthor
+{
+    string name, email;
+    string toString()
+    {
+        return "%s <%s>".format(name, email);
+    }
+}
+
+/// Options for finding authors
+struct FindConfig
+{
+    bool refreshTags; /// will query github.com for new tags
+    bool noMerges; // will ignore merge commits
+    bool showAllContributrs; // will ignore the revRange and show all contributors
+    string cwd; // working directory (should be tools)
+    string mailmapFile; // location to the .mailmap file
+}
+
+/**
+Search all git commit messages within revRange of all D repositories
+Returns: Array that maps each git `Author: ...` line to a GitAuthor
+*/
+auto findAuthors(string revRange, FindConfig config)
+{
+    Appender!(GitAuthor[]) authors;
+    int commits;
+    foreach (repo; ["dmd", "druntime", "phobos", "dlang.org", "tools", "installer"]
+             .map!(r => buildPath(config.cwd, "..", r)))
+    {
+        if (config.refreshTags)
+        {
+            auto cmd = ["git", "-C", repo, "fetch", "--tags", "https://github.com/dlang/" ~ repo.baseName,
+                               "+refs/heads/*:refs/remotes/upstream/*"];
+            auto p = pipeProcess(cmd, Redirect.stdout);
+            enforce(wait(p.pid) == 0, "Failed to execute '%(%s %)'.".format(cmd));
+        }
+
+        auto cmd = ["git", "-c", "mailmap.file=%s".format(config.mailmapFile), "-C", repo, "log", "--use-mailmap", "--pretty=format:%aN|%aE"];
```

Comment:
`git -C` only works with newer versions of git

Reply:
[git 1.8.3](https://github.com/git/git/blob/35f6318d44379452d8d33e880d8df0267b4a0cd0/Documentation/RelNotes/1.8.5.txt#L115) is an ancient dinosaur (more than four years old) :wink:

Reply:
std.process functions allow you to specify a working directory anyway.